test/lit/passes/gufa-cast-all-exact.wast

```diff
@@ -0,0 +1,54 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py and should not be edited.
+
+;; Check that exact casts are only added when custom descriptors are enabled.
+
+;; RUN: wasm-opt %s -all                              --gufa-cast-all -S -o -  | filecheck %s
+;; RUN: wasm-opt %s -all --disable-custom-descriptors --gufa-cast-all -S -o -  | filecheck %s --check-prefix=NO_CD
+
+(module
+  ;; CHECK:      (type $foo (struct))
+  ;; NO_CD:      (type $foo (struct))
+  (type $foo (struct))
+
+  ;; CHECK:      (import "" "" (global $g (ref (exact $foo))))
+  ;; NO_CD:      (import "" "" (global $g (ref (exact $foo))))
+  (import "" "" (global $g (ref (exact $foo))))
+
+  ;; CHECK:      (func $callee (type $1) (result (ref $foo))
+  ;; CHECK-NEXT:  (global.get $g)
+  ;; CHECK-NEXT: )
+  ;; NO_CD:      (func $callee (type $1) (result (ref $foo))
+  ;; NO_CD-NEXT:  (global.get $g)
+  ;; NO_CD-NEXT: )
+  (func $callee (result (ref $foo))
+    (global.get $g)
```

Comment:
Could this be `struct.new $foo` rather than global.get + a global?

Reply:
Eventually, but `struct.new` is not typed as exact yet.